client-lib/src/main/java/com/ibm/ws/repository/transport/model/WlpInformation.java

```diff
@@ -70,6 +70,22 @@
     private String mainAttachmentSHA256;
     private String genericRequirements;
     private String packagedJava;
+    private Collection<RequireFeatureWithTolerates> requiredFeaturesWithTolerates;
+
+    public Collection<RequireFeatureWithTolerates> getRequiredFeaturesWithTolerates() {
+        return requiredFeaturesWithTolerates;
+    }
+
+    public void setRequiredFeaturesWithTolerates(Collection<RequireFeatureWithTolerates> requiredFeaturesWithTolerates) {
+        this.requiredFeaturesWithTolerates = requiredFeaturesWithTolerates;
+    }
+
+    public void addRequiredFeaturesWithTolerates(RequireFeatureWithTolerates feature) {
+        if (this.requiredFeaturesWithTolerates == null) {
+            this.requiredFeaturesWithTolerates = new HashSet<RequireFeatureWithTolerates>();
+        }
+        this.requiredFeaturesWithTolerates.add(feature);
```

Comment:
You're using a `HashSet` to hold `RequireFeatureWithTolerates`, which suggests you want set semantics on the field.

However, the equality function for `RequireFeatureWithTolerates` includes both the feature name and the list of tolerated versions, so adding the same feature multiple times with different tolerated versions will have different behaviour to calling `addRequireFeature()` with the same feature twice.

Reply:
Changed so that only one entry will be added for each feature, so effectively making the set semantics based of just the feature name, rather than the feature name and the tolerates info